sys/net/routing/rpl/Makefile

```diff
@@ -1,16 +1,45 @@
-MODE = $(shell grep "RPL_DEFAULT_MOP RPL_NON_STORING_MODE" ../../include/rpl/rpl_config.h)
+# Default RPL_MOP values by IANA https://tools.ietf.org/html/rfc6550#page-129
+# RPL_MOP_NO_DOWNWARD_ROUTES  0x00
+# RPL_MOP_NON_STORING_MODE    0x01
+# RPL_MOP_STORING_MODE_NO_MC  0x02
+# RPL_MOP_STORING_MODE_MC     0x03
+
+# Set the RPL MOP iff it is matching a provided mode
+MODE :=
+
+ifeq (RPL_MOP_NO_DOWNWARD_ROUTES, $(RPL_MOP))
+	MODE := 0x00
+endif
+ifeq (RPL_MOP_NON_STORING_MODE, $(RPL_MOP))
+	MODE := 0x01
+endif
+ifeq (RPL_MOP_STORING_MODE_NO_MC, $(RPL_MOP))
+	MODE := 0x02
+endif
+ifeq (RPL_MOP_STORING_MODE_MC, $(RPL_MOP))
+	MODE := 0x03
+endif
+
+# If a valid RPL_MOP was provided, use it for RPL
 ifneq (,$(MODE))
+	CFLAGS += -DRPL_DEFAULT_MOP=$(MODE)
+ifeq (0x01,$(MODE))
```

Comment:
What's your opinion on moving the `DIRS += `assignments to the appropriate `ifeq .. endif`s above? This way we can remove the `ifneq .. endif` block in lines 24 - 30